client/test/error-telemetry.test.js

```diff
@@ -0,0 +1,45 @@
+jest.mock('fs');
+
+const assert         = require('assert');
+const fs             = require('fs');
+const ErrorTelemetry = require('../src/lib/error-telemetry');
+const mkdirp         = require('mkdirp');
+
+describe('Error Telemetry Logging', () => {
+  beforeEach(() => {
+    /* Make sure memfs is flushed every time */
+    fs.volume.reset();
+  });
+
+  describe('setup() call', () => {
+
+    // FIXME: only hackish, the end goal is definitely not to forward to another file
+    it('writing to essentials logging file should forward to another', done => {
```

Comment:
From a test structure perspective this is a bit hackish too. Please try to organize setup-type tasks into a `beforeEach` block rather than inside the `it` blocks

Generally I have been trying to keep the `it` blocks to only a few lines which include the _assertions_ rather than the entirety of the set up.

`beforeEach` for example, can be nested inside of nested `describe` blocks FWIW

Reply:
ack will do